File: src/main/java/kr/chuyong/springspigot/SpringSpigotApplication.java
package kr.chuyong.springspigot;

import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.scheduling.annotation.EnableScheduling;

@ComponentScan("kr.chuyong.springspigot")
@EnableAutoConfiguration
@EnableScheduling
public class SpringSpigotApplication {
    public SpringSpigotApplication(DataSourceAutoConfiguration configuration) {
        System.out.println("DB : " + configuration);
    }
}

File: src/main/kotlin/chuyong/springspigot/SpringSpigotApplication.kt
package chuyong.springspigot

import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
import org.springframework.context.annotation.ComponentScan
import org.springframework.scheduling.annotation.EnableScheduling

@ComponentScan("chuyong.springspigot")
@EnableAutoConfiguration
@EnableScheduling
class SpringSpigotApplication(
    private val db: DataSourceAutoConfiguration,
) {
    init {
        println(db)
    }
}
